nixos/modules/tasks/network-interfaces.nix

```diff
@@ -612,6 +657,93 @@ in
       };
     };
 
+    networking.wlanInterfaces = mkOption {
+      default = { };
+      example = {
+        wlan-adhoc0 = {
+          base = "wlp6s0";
+          type = "ibss";
+          mac = "02:00:00:00:00:01";
+        };
+      };
+      description =
+        ''
+          This option allows you to define vlan devices that tag packets
+          on top of a physical interface. The value of this option is an
+          attribute set. Each attribute specifies a vlan, with the name
+          specifying the name of the vlan interface.
```

Comment:
This description mentions vlans but not wlans, which is a bit confusing :-) I guess this is about defining virtual interfaces on top a single physical WLAN interface? Maybe that should be reflected in the option name (e.g. `networking.virtualWlanInterfaces` or so).